code/datums/loadout/item_representation/_item_representation.dm

```diff
@@ -66,19 +66,52 @@
 	for(var/atom/thing_in_content AS in item_to_copy.contents)
 		if(!isitem(thing_in_content))
 			continue
-		if(!is_savable_in_loadout(thing_in_content))
-			continue
 		item_representation_type = item2representation_type(thing_in_content.type)
 		contents += new item_representation_type(thing_in_content)
 
-/datum/item_representation/storage/instantiate_object(datum/loadout_seller/seller, master = null)
+/datum/item_representation/storage/instantiate_object(datum/loadout_seller/seller, master = null, mob/living/user)
 	. = ..()
 	if(!.)
 		return
 	var/obj/item/storage/storage = .
+	var/list/obj/item/starting_items = list()
+	for(var/obj/item/I AS in storage.contents)
+		starting_items[I.type] = starting_items[I.type] + 1
+	storage.delete_contents()
 	for(var/datum/item_representation/item_representation AS in contents)
-		var/obj/item/item_to_insert = item_representation.instantiate_object(seller)
+		if(!item_representation.bypass_vendor_check && starting_items[item_representation.item_type] > 0)
+			starting_items[item_representation.type] = starting_items[item_representation.item_type] - 1
+			item_representation.bypass_vendor_check = TRUE
+		var/obj/item/item_to_insert = item_representation.instantiate_object(seller, master, user)
 		if(!item_to_insert)
 			continue
 		if(storage.can_be_inserted(item_to_insert))
 			storage.handle_item_insertion(item_to_insert)
+			continue
+		item_to_insert.forceMove(user.loc)
+
+
+/**
+ * Allow to representate stacks of item of type /obj/item/stack
+ */
+/datum/item_representation/stack
+	///Amount of items in the stack
+	var/amount = 0
+
+/datum/item_representation/stack/New(obj/item/item_to_copy)
+	if(!item_to_copy)
+		return
+	if(!isitemstack(item_to_copy))
+		CRASH("/datum/item_representation/stack created from an item that is not a stack of items")
+	..()
+	var/obj/item/stack/stack_to_copy = item_to_copy
+	amount = stack_to_copy.amount
+
+/datum/item_representation/stack/instantiate_object(datum/loadout_seller/seller, master = null, mob/living/user)
+	if(seller && !bypass_vendor_check && !buy_item_in_vendor(item_type, seller, user) && !buy_stack(item_type, seller, user, amount))
```

Comment:
stacks are a bit special, since you can buy small parts in vendors, and then combine it.

This problem arises for metal, plasteel and metal